src/reconcile/rec_write.c

```diff
@@ -2943,6 +2943,7 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
               F_ISSET(r, WT_REC_IN_MEMORY) ||
                 (F_ISSET(r, WT_REC_EVICT) && r->leave_dirty && r->multi->supd_entries != 0),
               "Attempting a 1-for-1 page swap when there are still updates to write");
+            __wt_atomic_addv16(&ref->ref_changes, 1);
```

Comment:
We should increase the count for all the cases here for the current ref. Not just for 1 to to case.